tests/tests/test_end_to_end_api_test.py

```diff
@@ -0,0 +1,176 @@
+
+                    assessment_result_flag = assessment_response and assessment_response.status_code == 200
+                    assessment_id = assessment_response.json().get('id') if assessment_result_flag else None
+
+                    test_api_obj.log_result(
+                        assessment_result_flag,
+                        positive=f"Successfully retrieved assessment ID: {assessment_id}",
+                        negative=f"Failed to retrieve assessment ID. Response: {assessment_response.json() if assessment_response else assessment_response}."
+                    )
+
+                    if assessment_result_flag:
+                        rating_payload = {
+                            'capability_assessment_id': assessment_id,
+                            'rating': rating_details,
+                            'comment': 'Retrieving rating details'
+                        }
+
+                        rating_response = test_api_obj.submit_ratings(assessment_id=assessment_id, rating_details=rating_payload, auth_details=auth_details)
+
+                        rating_result_flag = rating_response and rating_response.status_code == 200
+                        test_api_obj.log_result(
+                            rating_result_flag,
+                            positive=f"Successfully submitted rating with details: {rating_response.json()}",
+                            negative=f"Failed to submit rating with response: {rating_response.json() if rating_response else rating_response}."
+                        )
+
+            # Delete ACC model
+            if acc_model_result_flag:
+                delete_response = test_api_obj.delete_acc_model(acc_model_id=acc_model_id, auth_details=auth_details)
+                delete_result_flag = delete_response and delete_response.status_code == 200
+
+                test_api_obj.log_result(
+                    delete_result_flag,
+                    positive=f"Successfully deleted ACC model with ID: {acc_model_id}",
+                    negative=f"Failed to delete ACC model. Response: {delete_response.json() if delete_response else delete_response}."
+                )
+
+            # Delete attributes
+            if attribute_result_flag:
+                delete_response = test_api_obj.delete_attribute(attribute_id=attribute_id, auth_details=auth_details)
+                delete_result_flag = delete_response and delete_response.status_code == 200
+
+                test_api_obj.log_result(
+                    delete_result_flag,
+                    positive=f"Successfully deleted attribute with ID: {attribute_id}",
+                    negative=f"Failed to delete attribute. Response: {delete_response.json() if delete_response else delete_response}."
+                )
+
+        # Update pass/fail counters
+        expected_pass = test_api_obj.total
+        actual_pass = test_api_obj.passed
+
+        # Write test summary
+        test_api_obj.write_test_summary()

```

Comment:
When I executed the test, the Delete operations failed for Attribute and acc_model. However, the failed checks were not displayed. Would be good to do that.
It just showed:
Total number of checks=7
2024-12-05 10:30:17.654 | INFO     | /home/sravanti/temp/acc-model-app/tests/utils/results.py::write_test_summary | Total number of checks passed=7


2024-12-05 10:30:17.656 | INFO     | /home/sravanti/temp/acc-model-app/tests/utils/results.py::write_test_summary | Total number of mini-checks=0
2024-12-05 10:30:17.659 | INFO     | /home/sravanti/temp/acc-model-app/tests/utils/results.py::write_test_summary | Total number of mini-checks passed=0
PASSED

But the delete operations failed. So there need to be checks for that too? And displayed in the summary. Else it is going unnoticed.